subprojects/vinumc/str.c

```diff
@@ -12,16 +12,28 @@ void put_str(struct str *to, char *from) {
 	VEC_PUT(to, '\0');
 }
 
-void put_start_blank_reduced_str(struct str *to, char *from) {
-	size_t len = strlen(from);
+void put_blank_reduced_str(struct str *to, char *from, bool skip_left, bool skip_right) {
 	if (to->len > 0) {
 		VEC_POP(to);
 	}
 
+	size_t start = 0;
+	size_t end = strlen(from);
+	if (skip_right) {
+		while (end > 0 && is_blank_char(from[end - 1])) {
+			end--;
+		}
+	}
+	if (skip_left) {
+		while (start < end && is_blank_char(from[start])) {
+			start++;
+		}
+	}
```

Comment:
couldn't this been added when you first created the function?

Reply:
I don't understand what you mean. Could you elaborate?